CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+
+[I believe the purpose here is to show that anyone ignoring the return value 
+of a function is doing so intentionally, but I'm considering pulling it anyway
+because it's a little weird.  Thoughts?]
+
+**Order of Operations**
+
+Expressions using two or more operators of different precedence must use
+parentheses to clarify the binding of operands to operators when the parsing
+order is not intuitive.
+
+**Types**
+
+For integer types in C and C++ code, use int, unsigned int, plain char and bool
+when possible, and don't use precise-width integer types in the absence of a
+specific sizing requirement.
+
+In other words, if a specific size is needed, a precise-width integer type
+(e.g., uint16_t) from `<stdint.h>` or equivalent must be used, and the following
+fundamental types must not be used: signed char, unsigned char, short, unsigned
+short, long, unsigned long, long long, unsigned long long. Exception: These
+types may be used to define precise-width types in `<stdint.h>` or equivalent.
+
+Only these standard C++ fundamental numeric types should be used:
+
+* `bool`
+* `char` (7-bit ASCII character)
+* `int` (32 bits, signed)
+* `unsigned` (32 bits, unsigned)
+* `size_t` (pointer size, 32 bits, unsigned)
+* `ptrdiff_t` (pointer difference size, 32 bits, signed)
+* `int8_t`
+* `int16_t`
+* `int32_t`
+* `int64_t`
+* `uint8_t`
+* `uint16_t`
+* `uint32_t`
+* `uint64_t`
+* `float`
+* `double`
+
+The C++ standard only guarantees the size of `int` and `unsigned` are at
+least 16 bits, but we assume they are 32 bits, as is common for 32-bit
+platforms.
+
+Signed values (`int`) should be prefered for representing quantities.  Unsigned
+values (`unsigned`) should not be used to "guarantee" that a function parameter
+or other variable is non-negative.  Assertions should be used instead.
+
+The C++ standard also does not specify the signedness of `char`, so is up to
+the compiler implementer to choose or make it a configurable option.  Do not
+rely on that choice.  Instead, use `int8_t`/`uint8_t` and use `char` only if
+referring to a character in a plain C-style string.
+
+Do not use `long double`.
+
+
+**Mutable**
+
+Using the mutable keyword requires justification.
+
+**Exceptions**
+
+*C++ exceptions must not be used.*
+
+[This was more of a hardware limitation for UASC, but it may make sense for us as 
+
+well. 
+Read up on the
+<a href="https://en.wikipedia.org/wiki/Design_by_contract">Design By Contract</a>
+programming style (possibly to be included as a separate document).]
+
+**dynamic_cast**
+
+Except for dynamic_cast, use of Run-Time Type Identification (RTTI) such as
+the typeid operator requires justification.
+
+**Friend**
+
+Using the friend keyword requires justification.
+
+**Protected**
+
+Using the protected keyword requires justification.
+
+**Initialization Lists**
+
+Constructor initializer lists must have the same order as the member
+data in the class definition.
```

Comment:
Also never change order, or add declaration of member variables in class without checking order of initializer lists.